### ISSUE TITLE
Click - exact element on click

### ISSUE THREAD

**Issue by mateuszbem** [Simonwep/selection#36]
I found common bug:
https://github.com/Simonwep/selection/issues/19
#### Do you want to request a **feature** or report a **bug**? 
Bug
#### What is the current behavior?
If user select by dragging - selections gets element from bbox (just touch).
Click behaviour doesn't work that way. You have to click exact element.
#### If the current behavior is a bug, please provide the steps to reproduce.
For example - unclosed circle. You have to click the line to set css class. But if you select by dragging rectangle it work's fine.
#### What is the expected behavior?
Click in Bbox of element triggers onSelect event.

![image](https://user-images.githubusercontent.com/37087603/57843037-e37afd00-77cd-11e9-9ceb-f3211fe89b36.png)

**Comment by Simonwep** [comment]
Sorry, I'm a little bit confused, do you mean `mode` support for single-click? :suspect:

**Comment by bemiks** [comment]
Nope, I mean - When you select by draggin, You have to just touch the element (it finds bbox of element).
When You single click - you have to click on exact line of it. When You click inside it doesn't work.

**Comment by Simonwep** [comment]
Okay, yeah - seems like the `click` event gets fired if the users clicks the drawed-area, not the boundings of it. So you mean that you need to click on the drawed area to select via single click, and only touch the bounding client rect if you use mousedrag?! 

If so, this isn't really a bug IMO. Just a behavior of how clicking and dragging gets interpreted. Detecting the drawed area, or vice versa would be way to computation-intensive regarding many elements.

**Comment by mateuszbem** [comment]
Indeed. So it's a feature :)

**Comment by bemiks** [comment]
Honestly - IMO it should be fixed. It look like that dragging and clicking works different way. 
Mode option should be for both way's. 'Touch' should work also for click.

**Comment by Simonwep** [comment]
If so, svg element e.g. paths would be also need to be taken into account since the `click` event get's only fired if the user actually clicked the path. So either the core would need to be rewritten, because obviously detecting the element, and calculating (for example) if the user touches a _path_ which can be anything and extremely complicated.

So yeah, changing it so the `mode` behavior also applies to `singleClick` would be the easiest. But since we're going really into detail, the hover and every other css-based effects which would indicate something _can be clicked_ would fail. 

Example:
![image](https://user-images.githubusercontent.com/30767528/58036832-4f41ca80-7b2c-11e9-98c8-02c03c671451.png)
The red area could be clicked and selected (using `mode: touch`), but no hover-effects would be applied since the user actually **doesn't** hovered the _element_. 
Moreover only the mode `touch` would make sence since you cant cover a element with your mouse and pressing exactly the center also makes no sence.

**Comment by Simonwep** [comment]
@bemiks I've added a `tapMode` option in 289a50b2cb9aedf582e2573f606440880775bc99 where you can use `touch` to get the same behavior as if you're making an area-selection :)

Other options may follow on request 😊
